src/main/java/io/cdap/plugin/salesforce/plugin/connector/SalesforceConnector.java

```diff
@@ -200,15 +208,21 @@ private List<StructuredRecord> listObjectDetails(String object, int limit) throw
     SObjectDescriptor sObjectDescriptor = SObjectDescriptor.fromQuery(query);
     SoapRecordToMapTransformer soapRecordToMapTransformer = new SoapRecordToMapTransformer();
     PartnerConnection partnerConnection = SalesforceConnectionUtil.getPartnerConnection(credentials);
-    QueryResult queryResult = partnerConnection.query(query);
+
+    QueryResult queryResult = PartnerConnectionRetryWrapper.queryWithRetry(partnerConnection, query);
```

Comment:
What is the suggested guideline here when calling any method on PartnerConnectionRetryWrapper, does the caller need to have a catch Exception block ? and why ?